src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/BuildOptions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.MSBuild
+{
+    [Serializable]
+    internal class BuildOptions : IHostBuildOptions
```

Comment:
The translators that convert build options into parse and compilation options exist in portable assembly w/o serializable attribute.

Ideally, the build options and translator would go away in favor of specifying everything as command line args and letting the command line compiler do the translation for use.